[devnet] Running create-account on fresh aztec node throws error

Setting up a fresh node with:

```
ETHEREUM_HOST='http://localhost:8545' DEPLOY_AZTEC_CONTRACTS=1 SEQ_PUBLISHER_PRIVATE_KEY=0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d yarn aztec start --node --sequencer --prover --archiver --pxe
```

And then running:

```
yarn aztec create-account
```

Triggers the error:

```
aztec:simulator:acvm [ERROR] Error in oracle callback storageRead: Invalid block number to get world state for: 0 +0ms
```

Running another command like `bootstrap` that advances the chain a bit fixes this.

I am getting the same error when trying to `bootstrap-network`:

```
  aztec:cli [ERROR] Error in command execution +0ms
  aztec:cli [ERROR] Error: (JSON-RPC PROPAGATED) Invalid block number to get world state for: 0
Error: (JSON-RPC PROPAGATED) Invalid block number to get world state for: 0
    at axiosFetch (file:///usr/src/yarn-project/cli/dest/client.js:36:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async request (file:///usr/src/yarn-project/foundation/dest/json-rpc/client/json_rpc_client.js:88:21)
    at async DeployAccountMethod.prove (file:///usr/src/yarn-project/aztec.js/dest/contract/base_contract_interaction.js:21:19)
    at async file:///usr/src/yarn-project/aztec.js/dest/contract/base_contract_interaction.js:35:36
    at async DeployAccountSentTx.waitForReceipt (file:///usr/src/yarn-project/aztec.js/dest/contract/sent_tx.js:86:24)
    at async DeployAccountSentTx.wait (file:///usr/src/yarn-project/aztec.js/dest/contract/sent_tx.js:47:25)
    at async DeployAccountSentTx.wait (file:///usr/src/yarn-project/aztec.js/dest/account_manager/deploy_account_sent_tx.js:26:25)
    at async DeployAccountSentTx.getWallet (file:///usr/src/yarn-project/aztec.js/dest/account_manager/deploy_account_sent_tx.js:17:25)
    at async bootstrapDevnet (file:///usr/src/yarn-project/cli/dest/cmds/devnet/bootstrap_devnet.js:13:20) +0ms
```

Is that expected? Version: 0.47.1.

--- a/yarn-project/aztec-node/src/aztec-node/server.ts
+++ b/yarn-project/aztec-node/src/aztec-node/server.ts
@@ -805,7 +805,7 @@ export class AztecNodeService implements AztecNode {
    * @returns An instance of a committed MerkleTreeOperations
    */
   async #getWorldState(blockNumber: L2BlockNumber) {
-    if (typeof blockNumber === 'number' && blockNumber < INITIAL_L2_BLOCK_NUM) {
+    if (typeof blockNumber === 'number' && blockNumber < INITIAL_L2_BLOCK_NUM - 1) {
       throw new Error('Invalid block number to get world state for: ' + blockNumber);
     }
 

--- a/yarn-project/end-to-end/src/e2e_block_building.test.ts
+++ b/yarn-project/end-to-end/src/e2e_block_building.test.ts
@@ -5,6 +5,7 @@ import {
   ContractDeployer,
   ContractFunctionInteraction,
   type DebugLogger,
+  Fq,
   Fr,
   L1NotePayload,
   type PXE,
@@ -306,6 +307,26 @@ describe('e2e_block_building', () => {
       // will likely not be useful when complete.
       // const decryptedLogs = encryptedLogs.map(l => TaggedNote.decryptAsIncoming(l.data, keys.masterIncomingViewingSecretKey));
     }, 60_000);
+  });
+
+  describe('regressions', () => {
+    afterEach(async () => {
+      if (teardown) {
+        await teardown();
+      }
+    });
+
+    // Regression for https://github.com/AztecProtocol/aztec-packages/issues/7537
+    it('sends a tx on the first block', async () => {
+      ({ teardown, pxe, logger, aztecNode } = await setup(0, {
+        minTxsPerBlock: 0,
+        sequencerSkipSubmitProofs: true,
+        skipProtocolContracts: true,
+      }));
+
+      const account = getSchnorrAccount(pxe, Fr.random(), Fq.random(), Fr.random());
+      await account.waitSetup();
+    });
   });
 });
 

--- a/yarn-project/end-to-end/src/fixtures/utils.ts
+++ b/yarn-project/end-to-end/src/fixtures/utils.ts
@@ -281,6 +281,8 @@ type SetupOptions = {
   stateLoad?: string;
   /** Previously deployed contracts on L1 */
   deployL1ContractsValues?: DeployL1Contracts;
+  /** Whether to skip deployment of protocol contracts (auth registry, etc) */
+  skipProtocolContracts?: boolean;
 } & Partial<AztecNodeConfig>;
 
 /** Context for an end-to-end test as returned by the `setup` function */
@@ -387,24 +389,26 @@ export async function setup(
 
   const { pxe } = await setupPXEService(aztecNode!, pxeOpts, logger);
 
-  logger.verbose('Deploying key registry...');
-  await deployCanonicalKeyRegistry(
-    new SignerlessWallet(pxe, new DefaultMultiCallEntrypoint(config.l1ChainId, config.version)),
-  );
-
-  logger.verbose('Deploying auth registry...');
-  await deployCanonicalAuthRegistry(
-    new SignerlessWallet(pxe, new DefaultMultiCallEntrypoint(config.l1ChainId, config.version)),
-  );
-
-  if (enableGas) {
-    logger.verbose('Deploying Fee Juice...');
-    await deployCanonicalFeeJuice(
+  if (!config.skipProtocolContracts) {
+    logger.verbose('Deploying key registry...');
+    await deployCanonicalKeyRegistry(
       new SignerlessWallet(pxe, new DefaultMultiCallEntrypoint(config.l1ChainId, config.version)),
     );
+
+    logger.verbose('Deploying auth registry...');
+    await deployCanonicalAuthRegistry(
+      new SignerlessWallet(pxe, new DefaultMultiCallEntrypoint(config.l1ChainId, config.version)),
+    );
+
+    if (enableGas) {
+      logger.verbose('Deploying Fee Juice...');
+      await deployCanonicalFeeJuice(
+        new SignerlessWallet(pxe, new DefaultMultiCallEntrypoint(config.l1ChainId, config.version)),
+      );
+    }
   }
 
-  const wallets = await createAccounts(pxe, numberOfAccounts);
+  const wallets = numberOfAccounts > 0 ? await createAccounts(pxe, numberOfAccounts) : [];
   const cheatCodes = CheatCodes.create(config.l1RpcUrl, pxe!);
 
   const teardown = async () => {

